src/Interpreters/Context.cpp

```diff
@@ -696,6 +696,41 @@ void Context::setUser(const String & name, const String & password, const Poco::
     setSettings(*access->getDefaultSettings());
 }
 
+void Context::setUserSecure(const String & name, const Poco::Net::SocketAddress & address)
```

Comment:
Too much of copy-pasted code from `Context::setUser()`. Could you please make some common private function instead?

Reply:
I though about it, but it is kind of in the middle, and I was afraid of making functions too specific, ok will look deeper